source/blog/2016-04-25-sso.html.md

```diff
@@ -0,0 +1,154 @@
+
+So I wanted a oVirt installation that will use kerberos for API authentication. For the web ui, kerberos is not always the best solution, so I wanted to integrated it in our CAS.
+
+The Apache part was easy to setup. It needs an external module, auth_cas_module, that can be found at [Apache's CAS module](https://wiki.jasig.org/display/CASC/mod_auth_cas). It builds without special tweaks with
+
+    ./configure
+    make
+    make install
+
+I will show only subpart of the whole Apache setup and only authentication related part
+
+    # The CAS modules
+    LoadModule authz_user_module      /usr/lib64/httpd/modules/mod_authz_user.so
+    # Needed because auth_cas_module forget to link openssl
+    LoadModule ssl_module            /usr/lib64/httpd/modules/mod_ssl.so
+    LoadModule auth_cas_module       /usr/lib64/httpd/modules/mod_auth_cas.so
+
+    # For the kerberos authentication on the API
+    LoadModule auth_gssapi_module    /usr/lib64/httpd/modules/mod_auth_gssapi.so
+    LoadModule session_module        /usr/lib64/httpd/modules/mod_session.so
+    LoadModule session_cookie_module /usr/lib64/httpd/modules/mod_session_cookie.so
+
+    CASLoginURL https://sso/cas/login
+    CASValidateSAML On
+    CASValidateURL https://sso/cas/samlValidate
+
+    <VirtualHost *:443>
+        RequestHeader unset X-Remote-User early
+        <LocationMatch ^/api($|/)>
+            RewriteEngine on
+            RewriteCond %{LA-U:REMOTE_USER} ^(.*@DOMAIN)$
+            RewriteRule ^(.*)$ - [L,P,E=REMOTE_USER:%1]
+
+            RequestHeader set X-Remote-User %{REMOTE_USER}s
+
+            AuthType GSSAPI
+            AuthName "GSSAPI Single Sign On Login"
+            GssapiCredStore keytab:.../httpd.keytab
+            Require valid-user
+
+            GssapiUseSessions On
+            Session On
+            SessionCookieName ovirt_gssapi_session path=/private;httponly;secure;
+        </LocationMatch>
+        <LocationMatch ^/(ovirt-engine($|/)|RHEVManagerWeb/|OvirtEngineWeb/|ca.crt$|engine.ssh.key.txt$|rhevm.ssh.key.txt$)>
+            AuthType CAS
+            Require valid-user
+            CASAuthNHeader X-Remote-User
+        </LocationMatch>
+    </VirtualHost>
+
+The authn file ``/etc/ovirt-engine/extensions.d/apachesso-authn.properties`` is :
+
+ovirt.engine.extension.name = apachesso-authn
+ovirt.engine.extension.bindings.method = jbossmodule
+ovirt.engine.extension.binding.jbossmodule.module = org.ovirt.engine-extensions.aaa.misc
+ovirt.engine.extension.binding.jbossmodule.class = org.ovirt.engineextensions.aaa.misc.http.AuthnExtension
+ovirt.engine.extension.provides = org.ovirt.engine.api.extensions.aaa.Authn
+ovirt.engine.aaa.authn.profile.name = apachesso
+ovirt.engine.aaa.authn.authz.plugin = DOMAIN-authz
+config.artifact.name = HEADER
+config.artifact.arg = X-Remote-User
```

Comment:
^ Here you miss indent, so properties file content is not interpreted as code example. Would be nice if it will be.